annif/rest.py

```diff
@@ -248,6 +258,11 @@ def learn(
     try:
         corpus = _documents_to_corpus(body, project.subjects)
         project.learn(corpus)
+    except OperationFailedException as err:
+        if str(err) == "Learning not enabled for project":
+            return learning_not_enabled_error(project_id)
+        else:
+            return server_error(err)
```

Comment:
I [added a test](https://github.com/NatLibFi/Annif/pull/843/commits/0fe8f0b079ebad7db786c94ef16d59032df1539e) with mocking just to have test coverage also on this line (but in reality this line should not be reached, but this could change in the future?). Not very good coding?

Reply:
The test with mocking is no more needed when using the new NotEnabledException.